apis/securesourcemanager/v1alpha1/securesourcemanagerrepository_types.go

```diff
@@ -0,0 +1,111 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var SecureSourceManagerRepositoryGVK = GroupVersion.WithKind("SecureSourceManagerRepository")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SecureSourceManagerRepositorySpec defines the desired state of SecureSourceManagerRepository
+// +kcc:proto=google.cloud.securesourcemanager.v1.Repository
+type SecureSourceManagerRepositorySpec struct {
+	/* Immutable. The Project that this resource belongs to. */
+	// +required
+	ProjectRef *refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Location of the instance. */
+	Location string `json:"location"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The SecureSourceManagerRepository name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable. The name of the instance in which the repository is hosted, formatted as
+	// `projects/{project_number}/locations/{location_id}/instances/{instance_id}`
+	// +required
+	InstanceRef *SecureSourceManagerInstanceRef `json:"instanceRef,omitempty"`
+
+	// Input only. Initial configurations for the repository.
+	InitialConfig *Repository_InitialConfig `json:"initialConfig,omitempty"`
+}
+
+// SecureSourceManagerRepositoryStatus defines the config connector machine state of SecureSourceManagerRepository
+type SecureSourceManagerRepositoryStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the SecureSourceManagerRepository resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *SecureSourceManagerRepositoryObservedState `json:"observedState,omitempty"`
+}
+
+// SecureSourceManagerRepositoryObservedState is the state of the SecureSourceManagerRepository resource as most recently observed in GCP.
+type SecureSourceManagerRepositoryObservedState struct {
+	// // Output only. Create timestamp.
+	// CreateTime *string `json:"createTime,omitempty"`
```

Comment:
Why this is commented out?

Reply:
I noticed it was commented out in `securesourcemanagerinstance_types.go`. Is it supposed to be? And is the observed state used to do any updates?

Reply:
I do not know why it's commented out in instance, maybe we can just ignore them for now. We can always add them as needed. 

`observed state` shows server generated(output only) fields, they will not be updated by user configuration.

Reply:
Ah I see, sounds fine then.